eos-metapackage

```diff
@@ -80,7 +80,37 @@ sub subst_depends {
 
 	# Allow the toplevel file to be missing (e.g., -recommends)
 	@deps = get_depends("$depfile") if -e "$depfile";
-	addsubstvar($package, $var, join(', ', @deps));
+
+	if ($ENV{EOS_META_CHECK}) {
+		verbose_print("checking all packages in $depfile are installable");
+		my @cmd;
+		my $arch = hostarch();
+
+		foreach my $dep (@deps) {
```

Comment:
I've never tried it, but you could try using `Dpkg::Deps::dep_parse($dep, reduce_restrictions => 1)`. You'd still have to filter out dependencies from this package, but it would allow you to cleanly manage the arch specific dependencies. See `dpkg-checkbuilddeps` and `dpkg-shlibdeps` for examples. You'd probably want to add a build dependency on `libdpkg-perl` even though it's implicitly pulled in by `debhelper`.

That said, if you've had enough of this I think it's probably fine to assume our simple usage matches the parsing below.

Reply:
Something was actually wrong in my hand-rolled parsing – I caught it excluding a `[!armhf]` package on my `amd64` system. It was pretty easy to swap in `Dpkg::Deps::deps_parse` and it made the surrounding code simpler. I also worked out how to add a command-line parameter rather than relying on a magic environment variable.